src/main/java/com/google/devtools/build/lib/analysis/ConfiguredTargetFactory.java

```diff
@@ -304,6 +308,49 @@ private ConfiguredTarget createRule(
                     prerequisiteMap.values()))
             .build();
 
+    if (!"toolchain".equals(rule.getRuleClass())) {
+      if (ruleContext.attributes().has("target_compatible_with")) {
+        Iterable<ConstraintValueInfo> constraintValues =
+            PlatformProviderUtils.constraintValues(
+                ruleContext.getPrerequisites("target_compatible_with", TransitionMode.DONT_CHECK));
+        for (ConstraintValueInfo cvi : constraintValues) {
+          if (!ruleContext.targetPlatformHasConstraint(cvi)) {
+            Artifact symlink = ruleContext.createOutputArtifact();
+
+            ruleContext.registerAction(
+                SymlinkAction.toAbsolutePath(
+                    ruleContext.getActionOwner(),
+                    PathFragment.create("/bin/false"),
+                    symlink,
+                    "Faking out " + ruleContext.getLabel()));
```

Comment:
Registering this "dummy" action silenced a lot of the previous errors bazel was throwing. It's not pretty. The symlink action was the simplest one I could find ad the time. Will have to find a different approach to this.